src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -57,6 +57,66 @@ private static unsafe float MathFMinMagnitude(float x, float y)
             float ax = MathF.Abs(x), ay = MathF.Abs(y);
             return (ax < ay) || float.IsNaN(ax) || (ax == ay && *(int*)&x < 0) ? x : y;
         }
+
+        private static unsafe int SingleToInt32(float f) => *(int*)&f;
+
+        private static unsafe float Int32ToSingle(int i) => *(float*)&i;
+
+        private static float AnotherSingleNaN = Int32ToSingle(-8388607);
+
+        /// <summary>Loads a variety of special values (e.g. NaN) into random positions in <paramref name="x"/>.</summary>
+        private static void SetSpecialValues(Span<float> x)
+        {
+            // NaN
+            x[s_random.Next(x.Length)] = float.NaN;
+            x[s_random.Next(x.Length)] = AnotherSingleNaN;
+
+            // +Infinity, -Infinity
+            x[s_random.Next(x.Length)] = float.PositiveInfinity;
+            x[s_random.Next(x.Length)] = float.NegativeInfinity;
+
+            // +Zero, -Zero
+            x[s_random.Next(x.Length)] = +0.0f;
+            x[s_random.Next(x.Length)] = -0.0f;
+
+            // +Epsilon, -Epsilon
+            x[s_random.Next(x.Length)] = +float.Epsilon;
+            x[s_random.Next(x.Length)] = -float.Epsilon;
```

Comment:
`Epsilon` is the smallest `subnormal`, (bit pattern `0x0000_0001` and `0x8000_0001`).

It's also interesting to cover the:
* largest `subnormal` (bit pattern `0x007F_FFFF` and `0x807F_FFFF`)
* smallest `normal` (bit pattern `0x0080_0000` and `0x8080_0000`)
* largest `normal` (bit pattern `0x7F7F_FFFF` and `0x8F7F_FFFF` --- `MaxValue` and `MinValue`)